### PR TITLE
Add dag source code to dag details page

### LINKED ISSUE #43713: AIP-38 | Add Dag source code to dag details
Use autogenerated queries to show the dag source code. Show numbers for each log lines, support syntax highlighting depending on language, and wrap/unwrapping lines.

### PR DESCRIPTION
**Pull request by bbovenzi**
Closes: #43713 

Use dag source endpoint to add dag code to the new UI:

Screenshot 2024-11-01 at 4 59 27 PM
Screenshot 2024-11-01 at 4 59 34 PM

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/ui/src/pages/DagsList/Dag/Code/Code.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  // TODO: get default_wrap from config
+  const [wrap, setWrap] = useState(false);
+
+  const toggleWrap = () => setWrap(!wrap);
+  const { colorMode } = useColorMode();
+
+  const style = colorMode === "dark" ? oneDark : oneLight;
+
+  // wrapLongLines wasn't working with the prsim styles so we have to manually apply the style
+  if (style['code[class*="language-"]'] !== undefined) {
```
Comment (tirkarthi) [inline airflow/ui/src/pages/DagsList/Dag/Code/Code.tsx:75]: This looks like it doesn't fix the issue as I tried applying wrap but not sure if it's different to the fix applied here. There is a PR open in the repo but not merged.

Issue : https://github.com/react-syntax-highlighter/react-syntax-highlighter/issues/376
PR : https://github.com/react-syntax-highlighter/react-syntax-highlighter/pull/483

Tested on : Ubuntu Firefox 131.0.2 (64-bit)
Sample dag : 

```python
import pendulum
from datetime import datetime, timedelta, timezone

from airflow import DAG
from airflow.decorators import task

# This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment

with DAG(
    dag_id="wrap_dag",
    start_date=datetime(2024, 1, 1),
    end_date=datetime(2024, 12, 12),
    schedule=timedelta(minutes=1),
) as dag:

    @task
    def this_is_a_very_long_function_name_to_enable_wrapping(ds=None):
        """
        This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment
        """
        pass

    @task
    def this_is_a_very_long_function_name_to_enable_wrapping_1(ds=None):
        """
        This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment
        """
        pass

    @task
    def this_is_a_very_long_function_name_to_enable_wrapping_2(ds=None):
        """
        This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment This is a very long comment
        """
        pass


    this_is_a_very_long_function_name_to_enable_wrapping() >> this_is_a_very_long_function_name_to_enable_wrapping_1() >> this_is_a_very_long_function_name_to_enable_wrapping_2()
```

image

**Comment by tirkarthi** [comment]
Also legacy webserver used `yarn.lock` which is `pnpm-lock.yaml` in new UI which doesn't match below expressions and is considered as a remaining file causing all tests to run.

https://github.com/apache/airflow/blob/20e6517ed334e5d76015bd3e170933b25f9337b5/dev/breeze/src/airflow_breeze/utils/selective_checks.py#L192-L197

File: airflow/ui/src/pages/DagsList/Dag/Code/Code.tsx (hunk 1 above)
Comment (bbovenzi) [reply airflow/ui/src/pages/DagsList/Dag/Code/Code.tsx:75]: Yes, we needed to apply two fixes. I just added the custom renderer in that issue thread. Now its working correctly:

Screenshot 2024-11-05 at 11 47 12 AM
Screenshot 2024-11-05 at 11 47 20 AM

Comment (tirkarthi) [reply airflow/ui/src/pages/DagsList/Dag/Code/Code.tsx:75]: Thanks @bbovenzi, tested the latest commit and can confirm it's working fine.

**Review by pierrejeambrun — APPROVED** [review]
Looking good

**Review by tirkarthi — APPROVED** [review]
Thanks @bbovenzi

**Pull request merged**